src/sage/libs/singular/function.pyx

```diff
@@ -1474,29 +1473,29 @@ cdef inline call_function(SingularFunction self, tuple args, object R, bint sign
 
     currentVoice = NULL
     myynest = 0
-    errorreported = 0
-
-    while error_messages:
-        error_messages.pop()
+    start_catch_error()
 
     with opt_ctx: # we are preserving the global options state here
         if signal_handler:
-            sig_on()
-            _res = self.call_handler.handle_call(argument_list, si_ring)
-            sig_off()
+            try:
+                sig_on()
+                _res = self.call_handler.handle_call(argument_list, si_ring)
+                sig_off()
+            finally:
+                s = check_error()
         else:
             _res = self.call_handler.handle_call(argument_list, si_ring)
+            s = check_error()
 
     if myynest:
         myynest = 0
 
     if currentVoice:
         currentVoice = NULL
```

Comment:
I could not find any use of `currentVoice` within sage, do you know what this is?  (It does appear in `local/include/singular/Singular/sdb.h` and `local/include/singular/Singular/fevoices.h`)

Reply:
No idea, but given the naming, my best guess is "who is speaking to Singular" i.e. "where is Singular taking commands from". The example has `currentVoice=feInitStdin(NULL);` (read: stdin) and we certainly don't want it to read from stdin.